File: frontend/src/types/paper.ts
export interface PaperDetails {
    /** 研究动机 */
    motivation: string;
    /** 研究方法 */
    method: string;
    /** 实验结果 */
    result: string;
    /** 结论 */
    conclusion: string;
}

export interface PaperLinks {
    /** PDF链接 */
    pdf: string;
    /** Arxiv链接 */
    arxiv: string;
    /** HTML页面链接 */
    html: string;
}

export interface RawPaperMetadata {
    id: string;
    title: string;
    authors: string[];
    published_date: string;
    category: string[];
    abstract: string;
    links: PaperLinks;
    comment?: string;
}

export interface PaperAnalysis {
    tldr?: string;
    motivation?: string;
    method?: string;
    result?: string;
    conclusion?: string;
    tags: Record<string, string>; // Changed from string[] to Dict
}

export interface UserPaperState {
    paper_id: string;
    user_id: string;
    relevance_score: number;
    why_this_paper?: string;
    accepted: boolean;
    user_accepted: boolean;
    user_liked?: boolean | null;
    user_feedback?: string;
}

export interface PersonalizedPaper {
    meta: RawPaperMetadata;
    analysis?: PaperAnalysis;
    user_state?: UserPaperState;
}

// Alias for backward compatibility
export type Paper = PersonalizedPaper;

File: backend/schemas/paper.py
from typing import List, Optional, Dict
from pydantic import BaseModel

class PaperDetails(BaseModel):
    """论文详细信息模型"""
    motivation: Optional[str] = None
    method: Optional[str] = None
    result: Optional[str] = None
    conclusion: Optional[str] = None
    abstract: str
    tldr: Optional[str] = None

class PaperLinks(BaseModel):
    """论文链接模型"""
    pdf: str
    arxiv: str
    html: str

class DislikeReason(BaseModel):
    """不感兴趣原因模型"""
    reason: str
    tags: List[str]

class Paper(BaseModel):
    """论文核心模型"""
    id: str
    title: str
    authors: List[str]
    date: str
    category: str
    tldr: Optional[str] = None
    suggestion: Optional[str] = None
    tags: List[str] = []
    details: PaperDetails
    links: PaperLinks
    citationCount: int
    year: Optional[int] = None
    isLiked: Optional[bool] = None
    whyThisPaper: Optional[str] = None

class PaperAnalysis(BaseModel):
    """论文深度分析模型"""
    summary: str
    key_points: PaperDetails
    recommendation_reason: str
    score: float
